harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -675,20 +691,75 @@ class PyTorchTrial(det.Trial):
+
+    2. **Run forward and backward passes**.
+
+       We should call ``backward`` and ``step_optimizer`` provided by
+       `PyTorchTrialContext <pytorch-trial-context_>`_ in :meth:`train_batch`.
+       Note that we support arbitrary number of models, optimizers, and LR schedulers
+       and arbitrary orders of running forward and backward passes.
+
+    2. **Configure automatic mixed precision**.
+
+       We should call ``configure_apex_amp`` provided by
+       `PyTorchTrialContext <pytorch-trial-context_>`_ in the :meth:`__init__`.
+
+    3. **Clip gradients**.
+
+       We should pass a function into ``step_optimizer(optimizer, clip_grads=...)``
+       provided by `PyTorchTrialContext <pytorch-trial-context_>`_ in the :meth:`train_batch`.
+
+    Another deprecated method to do the above things is as follows:
+
+    1. Define the abstract methods :meth:`build_model`, :meth:`optimizer`,
+       :meth:`create_lr_scheduler` for the model, optimizer, and LR scheduler, which
+       initialized by Determined. Note this method doesn't support multiple models,
+       optimizers, and LR schedulers.
+
+    2. Configure the experiment configuration field ``optimization.mixed_precision`` for
+       automatic mixed precision.
+
+    2. Clip gradients by adding a callback with ``on_before_optimizer_step`` in
+       :meth:`build_callbacks`.
+
+
+    .. _pytorch-trial-migration-guide:
+
+    .. note::
+
+        The new method of defining a PyTorch trial is more powerful in allowing
+        multiple models, optimizers, and LR schedulers and more flexible in running
+        forward and backward passes on them in arbitrary orders.
+        You can only choose one of the two methods.
+
+        To migrate from the deprecated method to the new one, please
+        change the following places in your code:
+
+        1. Wrap models, optimizers, and LR schedulers in the :meth:`__init__` with the methods
+           ``Model``, ``Optimizer``, and ``LRScheduler`` that are provided by
+           `PyTorchTrialContext <pytorch-trial-context_>`_. At the same time,
+           remove the implementation of :meth:`build_model`, :meth:`optimizer`,
+           :meth:`create_lr_scheduler`.
+
+        2. If using automatic mixed precision (AMP), configure Apex AMP in the ``__init__``
+           with the method ``configure_apex_amp`` that is provided by
+           `PyTorchTrialContext <pytorch-trial-context_>`_.
+           At the same time, remove the experiment configuration ``optimizations.mixed_precision``.
+
+        3. Run backward passes on losses and step optimizers in :meth:`train_batch` with the methods
+           ``backward`` and ``step_optimizer`` provided by
+           `PyTorchTrialContext <pytorch-trial-context_>`_. Clip graidents by passing a function
+           to the ``clip_grads`` argument of ``step_optimizer``
+           while removing the ``PyTorchCallback`` counterpart in :meth:`build_callbacks`.
```

Comment:
non-blocking: can we make this a separate doc from PyTorchTrial?  I guess I'm not sure where we would put it, which is why I'm marking this as non-blocking.

Reply:
Made it a separate section.